drivers/sensor/pixart/pat9136/pat9136_stream.c

```diff
@@ -0,0 +1,431 @@
+{
+	struct pat9136_data *data = dev->data;
+	const struct pat9136_config *cfg = dev->config;
+
+	k_timer_start(&data->stream.timer.backup,
+		      K_MSEC(cfg->backup_timer_period),
+		      K_NO_WAIT);
+}
+
+static void start_drdy_cooldown_timer(const struct device *dev)
+{
+	struct pat9136_data *data = dev->data;
+	const struct pat9136_config *cfg = dev->config;
+
+	k_timer_start(&data->stream.timer.cooldown.timer,
+		      K_MSEC(cfg->cooldown_timer_period),
+		      K_NO_WAIT);
+}
+
+static void pat9136_complete_result(struct rtio *ctx,
+				    const struct rtio_sqe *sqe,
+				    void *arg)
+{
+	const struct device *dev = (const struct device *)arg;
+	struct pat9136_data *data = dev->data;
+	struct rtio_iodev_sqe *iodev_sqe = data->stream.iodev_sqe;
+	struct pat9136_encoded_data *edata = sqe->userdata;
+
+	data->stream.iodev_sqe = NULL;
+
+	edata->header.events.drdy = true &&
+				    data->stream.settings.enabled.drdy;
+	edata->header.events.motion = REG_MOTION_DETECTED(edata->motion) &&
+				      data->stream.settings.enabled.motion;
+	edata->header.channels = 0;
+
+	if ((data->stream.settings.enabled.drdy &&
+	      data->stream.settings.opt.drdy == SENSOR_STREAM_DATA_INCLUDE) ||
+	    (data->stream.settings.enabled.motion &&
+	      data->stream.settings.opt.motion == SENSOR_STREAM_DATA_INCLUDE)) {
+		edata->header.channels |= pat9136_encode_channel(SENSOR_CHAN_POS_DXYZ);
+		edata->header.channels |= pat9136_encode_channel(SENSOR_CHAN_POS_DXYZ_MM);
+	}
+
+	if (data->stream.settings.enabled.drdy) {
+		start_drdy_backup_timer(dev);
+	}
+
+	/** Attempt chip recovery if erratic behavior is detected  */
+	if (!REG_OBSERVATION_READ_IS_VALID(edata->observation)) {
+
+		LOG_WRN("CHIP OK register indicates issues. Attempting chip recovery: 0x%02X",
+			edata->observation);
+
+		rtio_iodev_sqe_err(iodev_sqe, -EAGAIN);
+	} else {
+		rtio_iodev_sqe_ok(iodev_sqe, 0);
+	}
+}
+
+static void pat9136_stream_get_data(const struct device *dev)
+{
+	struct pat9136_data *data = dev->data;
+	uint64_t cycles;
+	int err;
+
+	CHECKIF(!data->stream.iodev_sqe) {
+		LOG_WRN("No RTIO submission with an INT GPIO event");
+		return;
+	}
+
+	start_drdy_cooldown_timer(dev);
+
+	struct pat9136_encoded_data *buf;
+	uint32_t buf_len;
+	uint32_t buf_len_required = sizeof(struct pat9136_encoded_data);
+
+	err = rtio_sqe_rx_buf(data->stream.iodev_sqe,
+			      buf_len_required,
+			      buf_len_required,
+			      (uint8_t **)&buf,
+			      &buf_len);
+	__ASSERT(err == 0, "Failed to acquire buffer (len: %d) for encoded data: %d. "
+			   "Please revisit RTIO queue sizing and look for "
+			   "bottlenecks during sensor data processing",
+			   buf_len_required, err);
+
+	/** Still throw an error even if asserts are off */
+	if (err) {
+		struct rtio_iodev_sqe *iodev_sqe = data->stream.iodev_sqe;
+
+		LOG_ERR("Failed to acquire buffer for encoded data: %d", err);
+
+		data->stream.iodev_sqe = NULL;
+		rtio_iodev_sqe_err(iodev_sqe, err);
+		return;
+	}
+
```

Comment:
You can check for the number of acquirable sqe's here to perhaps save the check below

```c
__ASSERT(rtio_sqe_acquirable(data->rtio.ctx) > 7, "Not enough SQEs in the pool!");
```

Reply:
I agree this could work. However, I'd like to still checking the sqe's aren't null to keep it flexible: if in the future someone watns to perform more op's (or less less op's), you wouldn't have to come back to this assert.

LMK if you still think otherwise and I'm happy to write it this way.